File: 5_multi_level_inheritance.py
class Product:
    amount=None
    def __init__(self,amount) -> None:
        self.amount= amount
        
    def display(self):
        print("Products need to be sold")
        
class Food(Product):
    type= None
    def __init__(self, amount, type) -> None:
        self.type= type
        super().__init__(amount)
        
    def show(self):
        print("Food is important")
        
class Cake(Food):
    flavour= None
    def __init__(self, amount, type, flavour) -> None:
        self.flavour= flavour
        super().__init__(amount, type)
    
    def taste(self):
        print("Yummy")
        
blackForest= Cake(20, "Dessert", "Chocolate")
blackForest.display()
blackForest.show()
blackForest.taste()
print(blackForest.type)

File: 5_multi_level_inheritance.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

class Product{
    public:
    int amount;
    Product(int amount){
        this->amount= amount;
    }
    void display(){
        cout<<"Products are meant to be stolen"<<endl;
    }

};

class Food: public Product{
    public:
        string type;
        Food(int amount, string type): Product(amount){
            // this->amount= amount;
            this->type= type;
        }
        void eat(){
            cout<<"Food is important"<<endl;
        }
};

class Cake: public Food{
    public:
    string flavor;
    Cake(int amount, string type, string flavour): Food(amount, type){
        this->flavor= flavor;
    }
    void show(){
        cout<<"Yummy"<<endl;
    }
};

int main(){
    Cake blackForest(20, "Dessert", "Chocolate");
    blackForest.eat();
    blackForest.display();
    blackForest.show();

   
return 0;
}